File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/subscriptions/SubscriptionsDao.java
package com.github.dedis.popstellar.repository.database.subscriptions;

import androidx.room.*;

import io.reactivex.Completable;

@Dao
public interface SubscriptionsDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  Completable insert(SubscriptionsEntity subscriptionsEntity);

  @Query("SELECT * FROM subscriptions WHERE lao_id = :laoId")
  SubscriptionsEntity getSubscriptionsByLao(String laoId);
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/subscriptions/SubscriptionsDao.kt
package com.github.dedis.popstellar.repository.database.subscriptions

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import io.reactivex.Completable

/** Interface to query the table containing the subscriptions and server address for each lao. */
@Dao
interface SubscriptionsDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  fun insert(subscriptionsEntity: SubscriptionsEntity): Completable

  /**
   * It selects from the table the connection entity for a given lao.
   *
   * @param laoId lao identifier for which retrieving the subscriptions
   */
  @Query("SELECT * FROM subscriptions WHERE lao_id = :laoId")
  fun getSubscriptionsByLao(laoId: String): SubscriptionsEntity?
}
